src/utils/utils.ts

```diff
@@ -0,0 +1,9 @@
+export const groupBy = <T, K extends keyof any>(
+    list: T[],
+    getKey: (item: T) => K) =>
+list.reduce((previous, currentItem) => {
+    const group = getKey(currentItem);
+    if (!previous[group]) previous[group] = [];
+    previous[group].push(currentItem);
+    return previous;
+}, {} as Record<K, T[]>);
```

Comment:
I would suggest to improve formating
```suggestion
export const groupBy = <T, K extends keyof any>(
    list: T[],
    getKey: (item: T) => K
) => 
    list.reduce((previous, currentItem) => {
        const group = getKey(currentItem);
        if (!previous[group]) previous[group] = [];
        previous[group].push(currentItem);
        return previous;
    }, {} as Record<K, T[]>);
```

Reply:
Applied auto formatting